### PR TITLE
[RLlib] Quick-fix for default RLModules in combination with a user-provided config-sub-dict (instead of a full `DefaultModelConfig`).

### PR DESCRIPTION
**Pull request by sven1977**
Quick-fix for default RLModules in combination with a user-provided config-sub-dict (instead of a full `DefaultModelConfig`).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/dqn/dqn_rainbow_rl_module.py [hunk 1]
```diff
@@ -29,6 +31,11 @@
 class DQNRainbowRLModule(RLModule, InferenceOnlyAPI, TargetNetworkAPI):
     @override(RLModule)
     def setup(self):
+        # TODO (sven): Temporary fix (until we have figured out the final config
+        #  architecture for default models). If self.model_config is a dict (which
+        #  it should always be, make sure to merge it with the default config).
+        self.model_config = dataclasses.asdict(DefaultModelConfig()) | self.model_config
```
Comment (simonsays1980) [inline rllib/algorithms/dqn/dqn_rainbow_rl_module.py:37]: Does this mean that the user needs to do this everytime she defines a new module? We do here actually the same we did before with the `AlgorithmConfig.model_config` property.

File: rllib/algorithms/sac/sac_rl_module.py [hunk 2]
```diff
@@ -54,6 +56,11 @@ class SACRLModule(RLModule, InferenceOnlyAPI, TargetNetworkAPI):
 
     @override(RLModule)
     def setup(self):
+        # TODO (sven): Temporary fix (until we have figured out the final config
+        #  architecture for default models). If self.model_config is a dict (which
+        #  it should always be, make sure to merge it with the default config).
+        self.model_config = dataclasses.asdict(DefaultModelConfig()) | self.model_config
```
Comment (simonsays1980) [inline rllib/algorithms/sac/sac_rl_module.py:62]: I guess we need to add this also to the `BCRLModule` then.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Some general questions and maybe we need to add this also to BCRLModule

File: rllib/algorithms/sac/sac_rl_module.py (hunk 2 above)
Comment (simonsays1980) [reply rllib/algorithms/sac/sac_rl_module.py:62]: Also: We might need to add this to all the examples that are not derived from the algorithm modules.

Comment (sven1977) [reply rllib/algorithms/sac/sac_rl_module.py:62]: Great catch! Done ...

Comment (sven1977) [reply rllib/algorithms/sac/sac_rl_module.py:62]: For the non-default models, this is fine, b/c those are 100% user-driven and thus don't use the catalogs. The user decides, which keys are accessed in the `model_config` and only those need to be provided (or even have default values inside the custom RLModule code).

This is really only a problem for RLlib's default models.

File: rllib/algorithms/dqn/dqn_rainbow_rl_module.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/dqn/dqn_rainbow_rl_module.py:37]: > Does this mean that the user needs to do this everytime she defines a new module? We do here actually the same we did before with the `AlgorithmConfig.model_config` property

Nope, this is only(!) needed for RLlib's default models. See my other answer here. Users are completely free to request any keys they need in their custom model classes and then - of course - have to provide these keys in the `model_config` dict they pass into the `RLModuleSpec`. Users can also decide to use default values inside their `RLModule` classes for certain keys.

**Pull request merged**